<b>О датасете</b>

Набор данных Netflix содержит доход и количество платных подписчиков в зависимости от региона.

In [14]:
import pandas as pd
import plotly.express as px
import plotly.graph_objs as go
pd.set_option('display.max_columns', None)

In [15]:
df = pd.read_csv('D:/Work/Notebooks/Netflix Revenue.csv', sep = ',')
df

Date  Global Revenue  UACN Revenue  EMEA Revenue  LATM Revenue  \
0   31-03-2012       869791000           NaN           NaN           NaN   
1   30-06-2012       889163000           NaN           NaN           NaN   
2   30-09-2012       905089000           NaN           NaN           NaN   
3   31-12-2012       945239000           NaN           NaN           NaN   
4   31-03-2013      1023961000           NaN           NaN           NaN   
5   30-06-2013      1069372000           NaN           NaN           NaN   
6   30-09-2013      1105999000           NaN           NaN           NaN   
7   31-12-2013      1175230000           NaN           NaN           NaN   
8   31-03-2014      1270089000           NaN           NaN           NaN   
9   30-06-2014      1340407000           NaN           NaN           NaN   
10  30-09-2014      1409432000           NaN           NaN           NaN   
11  31-12-2014      1484728000           NaN           NaN           NaN   
12  31-03-2015      1573129000           NaN           NaN           NaN   
13  30-06-2015      1644694000           NaN           NaN           NaN   
14  30-09-2015      1738355000           NaN           NaN           NaN   
15  31-12-2015      1823333000           NaN           NaN           NaN   
16  31-03-2016      1957736000           NaN           NaN           NaN   
17  30-06-2016      2105204000           NaN           NaN           NaN   
18  30-09-2016      2290188000           NaN           NaN           NaN   
19  31-12-2016      2477541000           NaN           NaN           NaN   
20  31-03-2017      2636635000           NaN           NaN           NaN   
21  30-06-2017      2785464000           NaN           NaN           NaN   
22  30-09-2017      2984859000           NaN           NaN           NaN   
23  31-12-2017      3285755000           NaN           NaN           NaN   
24  31-03-2018      3700856000           NaN           NaN           NaN   
25  30-06-2018      3907270000           NaN           NaN           NaN   
26  30-09-2018      3999374000           NaN           NaN           NaN   
27  31-12-2018      4186841000           NaN           NaN           NaN   
28  31-03-2019      4520992000  2.256851e+09  1.233379e+09  6.304720e+08   
29  30-06-2019      4923116000  2.501199e+09  1.319087e+09  6.771360e+08   
30  30-09-2019      5244905000  2.621250e+09  1.428040e+09  7.414340e+08   
31  31-12-2019      5467434000  2.671908e+09  1.562561e+09  7.463920e+08   
32  31-03-2020      5767691000  2.702776e+09  1.723474e+09  7.934530e+08   
33  30-06-2020      6148286000  2.839670e+09  1.892537e+09  7.853680e+08   
34  30-09-2020      6435637000  2.933445e+09  2.019083e+09  7.893840e+08   
35  31-12-2020      6644442000  2.979505e+09  2.137158e+09  7.885220e+08   
36  31-03-2021      7163282000  3.170972e+09  2.343674e+09  8.366470e+08   
37  30-06-2021      7341777000  3.234643e+09  2.400480e+09  8.608820e+08   
38  30-09-2021      7483467000  3.257697e+09  2.432239e+09  9.152970e+08   
39  31-12-2021      7709318000  3.308788e+09  2.523426e+09  9.641500e+08   
40  31-03-2022      7867767000  3.350424e+09  2.561831e+09  9.989480e+08   
41  30-06-2022      7970141000  3.537863e+09  2.457235e+09  1.030234e+09   
42  30-09-2022      7925589000  3.601565e+09  2.375814e+09  1.023945e+09   
43  31-12-2022      7852053000  3.594791e+09  2.350135e+09  1.016846e+09   
44  31-03-2023      8161503000  3.608645e+09  2.517641e+09  1.070192e+09   

    APAC Revenue  UACN Members  EMEA  Members  LATM Members  APAC Members  \
0            NaN           NaN            NaN           NaN           NaN   
1            NaN           NaN            NaN           NaN           NaN   
2            NaN           NaN            NaN           NaN           NaN   
3            NaN           NaN            NaN           NaN           NaN   
4            NaN           NaN            NaN           NaN           NaN   
5            NaN           NaN            NaN      

Global Revenue: Общий доход.<br>
UACN : США и Канада<br>
EMEA : Европа, Средний Восток и Африка<br>
APAC : Азия-Тихоокеанический район<br>
LATM : Латинская Америка<br>
С 2019 года по сегодняшний день Netflix делит данные по регионам, до этого он делил на внутренний рынок и международный.<br>
До 2019 года количество глобальных пользователей было больше, потому что он резко потерял своих подписчиков.<br>
RPU - это доход на одного участника.<br>
Весь доход в долларах США.

In [16]:
#удаляем запятые в столбце Netflix Global Users, для лучшего восприятия
df['Netflix Global Users'] = df['Netflix Global Users'].replace(',','', regex=True)

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45 entries, 0 to 44
Data columns (total 21 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Date                         45 non-null     object 
 1   Global Revenue               45 non-null     int64  
 2   UACN Revenue                 17 non-null     float64
 3   EMEA Revenue                 17 non-null     float64
 4   LATM Revenue                 17 non-null     float64
 5   APAC Revenue                 17 non-null     float64
 6   UACN Members                 17 non-null     float64
 7   EMEA  Members                17 non-null     float64
 8   LATM Members                 17 non-null     float64
 9   APAC Members                 17 non-null     float64
 10  UACN RPU                     17 non-null     float64
 11  EMEA RPU                     17 non-null     float64
 12  LATM  RPU                    17 non-null     float64
 13  APAC  RPU             

In [18]:
df['Date'] = pd.to_datetime(df['Date'], format='%d-%m-%Y')
df['Netflix Global Users'] = df['Netflix Global Users'].astype(int)

In [19]:
fig = go.Figure()
fig.add_trace(go.Scatter(x=df['Date'], y=df['Netflix Global Users'], name='Всего пользователей'))
fig.update_layout(title='График общего количества пользователей',
                  xaxis_title='Дата',
                  yaxis_title='Кол-во пользователей',)
fig.show()

Очень странная статистика, возможно в какой-то момент Netflix поняли, что неправильно считают людей и разделили их на регионы и с марта 2019 начали подсчитывать людей по-новому.

In [20]:
#разделим на 2 дата фрейма, до разделения на региона в 2019 и после
df_old = df.dropna(subset=['Domestic Members'])
df_new = df.dropna(subset=['UACN Revenue'])
df_new = df_new.dropna(axis='columns')
df_old = df_old.dropna(axis='columns')

In [21]:
df_new

Date  Global Revenue  UACN Revenue  EMEA Revenue  LATM Revenue  \
28 2019-03-31      4520992000  2.256851e+09  1.233379e+09  6.304720e+08   
29 2019-06-30      4923116000  2.501199e+09  1.319087e+09  6.771360e+08   
30 2019-09-30      5244905000  2.621250e+09  1.428040e+09  7.414340e+08   
31 2019-12-31      5467434000  2.671908e+09  1.562561e+09  7.463920e+08   
32 2020-03-31      5767691000  2.702776e+09  1.723474e+09  7.934530e+08   
33 2020-06-30      6148286000  2.839670e+09  1.892537e+09  7.853680e+08   
34 2020-09-30      6435637000  2.933445e+09  2.019083e+09  7.893840e+08   
35 2020-12-31      6644442000  2.979505e+09  2.137158e+09  7.885220e+08   
36 2021-03-31      7163282000  3.170972e+09  2.343674e+09  8.366470e+08   
37 2021-06-30      7341777000  3.234643e+09  2.400480e+09  8.608820e+08   
38 2021-09-30      7483467000  3.257697e+09  2.432239e+09  9.152970e+08   
39 2021-12-31      7709318000  3.308788e+09  2.523426e+09  9.641500e+08   
40 2022-03-31      7867767000  3.350424e+09  2.561831e+09  9.989480e+08   
41 2022-06-30      7970141000  3.537863e+09  2.457235e+09  1.030234e+09   
42 2022-09-30      7925589000  3.601565e+09  2.375814e+09  1.023945e+09   
43 2022-12-31      7852053000  3.594791e+09  2.350135e+09  1.016846e+09   
44 2023-03-31      8161503000  3.608645e+09  2.517641e+09  1.070192e+09   

    APAC Revenue  UACN Members  EMEA  Members  LATM Members  APAC Members  \
28   319602000.0    66633000.0     42542000.0    27547000.0    12141000.0   
29   349494000.0    66501000.0     44229000.0    27890000.0    12942000.0   
30   382304000.0    67114000.0     47355000.0    29380000.0    14485000.0   
31   418121000.0    67662000.0     51778000.0    31417000.0    16233000.0   
32   483660000.0    69969000.0     58734000.0    34318000.0    19835000.0   
33   569140000.0    72904000.0     61483000.0    36068000.0    22492000.0   
34   634891000.0    73081000.0     62242000.0    36324000.0    23504000.0   
35   684609000.0    73936000.0     66698000.0    37537000.0    25492000.0   
36   762414000.0    74384000.0     68508000.0    37894000.0    26853000.0   
37   799480000.0    73951000.0     68696000.0    38658000.0    27875000.0   
38   834002000.0    74024000.0     70500000.0    38988000.0    30051000.0   
39   870705000.0    75215000.0     74036000.0    39961000.0    32632000.0   
40   916754000.0    74579000.0     73733000.0    39610000.0    33719000.0   
41   907719000.0    73283000.0     72966000.0    39624000.0    34799000.0   
42   889037000.0    73387000.0     73534000.0    39936000.0    36228000.0   
43   856711000.0    74296000.0     76729000.0    41699000.0    38023000.0   
44   933523000.0    74398000.0     77373000.0    41249000.0    39478000.0   

    UACN RPU  EMEA RPU  LATM  RPU  APAC  RPU  Netflix Global Users  
28     11.45     10.23       7.84       9.37             148863000  
29     12.52     10.13       8.14       9.29             151562000  
30     13.08     10.40       8.63       9.29             158334000  
31     13.22     10.51       8.18       9.07             167090000  
32     13.09     10.40       8.05       8.94             182856000  
33     13.25     10.50       7.44       8.96             192947000  
34     13.40     10.88       7.27       9.20             195151000  
35     13.51     11.05       7.12       9.32             203663000  
36     14.25     11.56       7.39       9.71             207639000  
37     14.54     11.66       7.50       9.74             209180000  
38     14.68     11.65       7.86       9.60             213563000  
39     14.78     11.64       8.14       9.26             221844000  
40     14.91     11.56       8.37       9.21             221641000  
41     15.95     11.17       8.67       8.83             220672000  
42     16.37     10.81       8.58       8.34             223085000  
43     16.23     10.43       8.30       7.69             230747000  
44     16.18     10.89       8.60       8.03             232498000

In [22]:
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['UACN Revenue'], name='Америка и Канада'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['EMEA Revenue'], name='Европа, Средний Восток и Африка'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['LATM Revenue'], name='Латинская Америка'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['APAC Revenue'], name='Азия-Тихоокеанический район'))
fig.update_layout(title='График доходов по регионам',
                  xaxis_title='Дата',
                  yaxis_title='Доход',
                  legend_orientation="h")
fig.show()

В Америке и Канаде самый большой доход (3.6 трлн `$`), у Европейского региона заметен небольшой спад под конец 2022 года, но к апрелю 2023 практически вернул себе свою позицию (2.5 трлн `$`). У Азии-Тихоокеанического района и Латинской Америки виден небольшой, но стабильный рост(1.07 трлн `$` и 933 млн `$`).

In [23]:
#не получалось вывести следующий график из-за того, что в названии столбца оказался лишний пробел
df_new = df_new.rename(columns={'EMEA  Members': 'EMEA Members'})

In [24]:
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['UACN Members'], name='Америка и Канада'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['EMEA Members'], name='Европа, Средний Восток и Африка'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['LATM Members'], name='Латинская Америка'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['APAC Members'], name='Азия-Тихоокеанический район'))
fig.update_layout(title='График количества пользователей по регионам',
                  xaxis_title='Дата',
                  yaxis_title='Кол-во пользователей',
                  legend_orientation="h")
fig.show()

В середине 2022 по количеству пользователей Европа, Средний Восток и Африка (77.373 млн пользователеей) обогнали Америку с Канадой(74.398 млн пользователей), думаю количество первых так и будет расти, потому что заметна очень продолжителльная стабильность по количеству пользователей в Северной Америке, так сказать, кто хотел, тот подключил.<br> 
В Латинской Америке заметна появляющаяся стабильность в количестве пользователей (41.249 млн пользователей). <br>
Азия и Тихоокеанский район очень хорошо набирает пользователей, за 4 года набрали около 30 млн пользователей (На момент анализа максимальное количество 39.478 млн пользователей).

In [25]:
df_new = df_new.rename(columns={'LATM  RPU': 'LATM RPU'})
df_new = df_new.rename(columns={'APAC  RPU': 'APAC RPU'})

In [26]:
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['UACN RPU'], name='Америка и Канада'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['EMEA RPU'], name='Европа, Средний Восток и Африка'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['LATM RPU'], name='Латинская Америка'))
fig.add_trace(go.Scatter(x=df_new['Date'], y=df_new['APAC RPU'], name='Азия-Тихоокеанический район'))
fig.update_layout(title='График доходов по регионам',
                  xaxis_title='Дата',
                  yaxis_title='Доход с человека',
                  legend_orientation="h")
fig.show()

В Америке с Канадой больше всего получают дохода с 1 человека в месяц (16.18 `$`).<br>
В Европе, Среднем Востоке и Африке 10.89 `$`.<br>
Латинская Америка 8.6 `$`.<br>
В Азии и Тихоокеанском районе заметен спад дохода с июля 2021 по январь 2023, затем пошел небольшой рост (8.03 `$`).